main.go

```diff
@@ -33,4 +86,47 @@ func printUsage() {
 		i++
 	}
 	fmt.Println()
+	fmt.Println(DetailedVersionString())
+}
+
+// reportStatus saves operation status to the status file for the extension
+// handler with the optional given message, if the given cmd requires reporting
+// status.
+//
+// If an error occurs reporting the status, it will be logged and returned.
+func reportStatus(ctx *log.Context, hEnv vmextension.HandlerEnvironment, t status.Type, c cmd, msg string) error {
+	if !c.shouldReportStatus {
+		ctx.Log("status", "not reported for operation (by design)")
+		return nil
+	}
+	s := status.NewStatus(t, c.name, statusMsg(c, t, msg))
+	seq, _ := strconv.Atoi(hEnv.SeqNo)
+	if err := s.Save(hEnv.HandlerEnvironment.StatusFolder, seq); err != nil {
```

Comment:
@ahmetalpbalkan Yes, make it an atomic move.  We had an issue with one of the marketplace templates where a large number of VMs are deployed.  I have a sample template to reproduce the issue reliably.